MekHQ/src/mekhq/campaign/finances/Finances.java

```diff
@@ -231,6 +234,122 @@ public void addLoan(Loan loan) {
 	}
 
 	public void newDay(Campaign campaign) {
+        DecimalFormat formatter = new DecimalFormat();
+        CampaignOptions campaignOptions = campaign.getCampaignOptions();
+        GregorianCalendar calendar = campaign.getCalendar();
+
+        // check for a new year
+        if (calendar.get(Calendar.MONTH) == 0 && calendar.get(Calendar.DAY_OF_MONTH) == 1) {
+            // clear the ledger
+            newFiscalYear(calendar.getTime());
+        }
+
+        // Handle contract payments
+        if (calendar.get(Calendar.DAY_OF_MONTH) == 1) {
+            for (Contract contract : campaign.getActiveContracts()) {
+                credit(contract.getMonthlyPayOut(), Transaction.C_CONTRACT,
+                        "Monthly payment for " + contract.getName(), calendar.getTime());
+                campaign.addReport("Your account has been credited for " + formatter.format(contract.getMonthlyPayOut())
+                        + " C-bills for the monthly payout from contract " + contract.getName());
+
+                if (campaignOptions.getUseAtB() && campaignOptions.getUseShareSystem()
+                        && contract instanceof AtBContract) {
+                    long shares = contract.getMonthlyPayOut() * ((AtBContract) contract).getSharesPct() / 100;
+                    if (debit(shares, Transaction.C_SALARY, "Shares payments for " + contract.getName(),
+                            calendar.getTime())) {
+                        campaign.addReport(formatter.format(shares) + " C-bills have been distributed as shares.");
+                    } else {
+                        /*
+                         * This should not happen, as the shares payment is less than the contract
+                         * payment that has just been made.
+                         */
+                        campaign.addReport(
+                                "<font color='red'><b>You cannot afford to pay shares!</b></font> Lucky for you that personnel morale is not yet implemented.");
```

Comment:
Can we tokenize the text in this method?

Reply:
Sure I guess, though it could also wait until the code that matters is actually implemented. This was just a refactor to move code out of Campaign.java and into Finances.java where it belongs (and also into smaller methods). But adding this text to the resource bundle is easy.

Reply:
Right, I picked a poor line for the comment. We can probably ignore text for "placeholder" mechanisms. I was thinking about the other text for features that have been implemented.

Reply:
I've added done the resource bundle work here. Another case of review by commit instead of the final result since I did another whitespace format commit and a couple of merge conflict commits too.